### DISCUSSION TITLE
Getting auth token from onModelBeforeCreate hook?

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#4593] [Q&A] [answered]
Hello,

I am in the middle of migrating from `onCollectionBeforeCreateRequest` and `onRecordBeforeCreateRequest` to `onModelBeforeCreate` for the cascading hook support.

I have two Pocketbase instances, the first one (source) has the following hook:

```js
onModelBeforeCreate((evt) => {
    const token = $apis.requestInfo(evt.httpContext).headers.authorization;

    const res = $http.send({
        method: "POST",
        body: JSON.stringify(evt),
        url: `http://localhost:8080/create`,
        headers: {
            "content-type": "application/json",
            "Authorization": `Bearer ${token}`,
        },
    });

    evt.model.set("status", res.statusCode);
});
```

Another one (destination):
```js
routerAdd("POST", "/create", (ctx) => {
    const dao = $app.dao().withoutHooks();
    const model = $apis.requestInfo(ctx).data.model;

    if ("schema" in model) { // new collection created 
        const collection = new Collection(model);
        dao.saveCollection(collection);
        return ctx.json(200, { id: collection.id });
    } else { // new record created
        const record = new Record(dao.findCollectionByNameOrId(model.collectionId), model);
        dao.saveRecord(record);
        return ctx.json(200, { id: record.id });
    }
}, $apis.requireAdminOrRecordAuth());
```

I want all of my custom endpoints to be protected by an auth token. I can get the token from `onCollectionBeforeCreateRequest` but not from `onModelBeforeCreate`.

Is it possible to get the token from `onModelBeforeCreate`?

**Comment by ganigeorgiev** [comment, marked as answer]
No, because the DB model hooks don't operate with the request context. Only the `*Request` hooks have `e.httpContext` field.

_Side-note: the middleware `$apis.requireAdminOrRecordAuth()` allows also regular user authentication to access your endpoint, not just admins, so if that's not intended you can replace it with `$apis.requireAdminAuth()`._

**Reply by AfdhalAtiffTan** [reply]
Thank @ganigeorgiev for the very quick reply.

I think I will try to use a reverse proxy as a workaround. 

Thanks again!
